### PR TITLE
LoopInfo: introduce Loop::getLocStr; unify debug output

### PR DESCRIPTION
**Pull request by artagnon**
Introduce a Loop::getLocStr stolen from LoopVectorize's static function getDebugLocString in order to have uniform debug output headers across LoopVectorize, LoopAccessAnalysis, and LoopDistribute. The motivation for this change is to have UpdateTestChecks recognize the headers and automatically generate CHECK lines for debug output, with minimal special-casing.

### CODE REVIEW

**Comment by artagnon** [comment]
Gentle ping. This patch should be straightforward.

File: llvm/test/Analysis/LoopAccessAnalysis/scalable-vector-regression-tests.ll [hunk 1]
```diff
@@ -24,7 +24,7 @@ loop.end:
   ret void
 }
 
-; CHECK-LABEL: LAA: Found a loop in regression_test_loop_access_scalable_typesize
+; CHECK-LABEL: 'regression_test_loop_access_scalable_typesize'
 ; CHECK: LAA: Bad stride - Scalable object:
```
Comment (RKSimon) [inline llvm/test/Analysis/LoopAccessAnalysis/scalable-vector-regression-tests.ll:28]: Should we be checking any extra lines after the label now?

Comment (artagnon) [reply llvm/test/Analysis/LoopAccessAnalysis/scalable-vector-regression-tests.ll:29]: Fixed, thanks.

**Comment by artagnon** [comment]
Gentle ping.

**Review by RKSimon — APPROVED** [review]
LGTM

File: llvm/lib/Analysis/LoopAccessAnalysis.cpp [hunk 2]
```diff
@@ -2403,6 +2403,8 @@ bool LoopAccessInfo::canAnalyzeLoop() {
     return false;
   }
 
+  LLVM_DEBUG(dbgs() << "LAA: Found a loop: " << TheLoop->getHeader()->getName()
```
Comment (fhahn) [inline llvm/lib/Analysis/LoopAccessAnalysis.cpp:2406]: Is this accurate? Might be better to say found an analyzable loop?

File: llvm/include/llvm/Analysis/LoopInfo.h [hunk 3]
```diff
@@ -385,6 +385,11 @@ class LLVM_EXTERNAL_VISIBILITY Loop : public LoopBase<BasicBlock, Loop> {
   /// Return the source code span of the loop.
   LocRange getLocRange() const;
 
+  /// Return a string containing the location of the loop (file name + line
```
Comment (fhahn) [inline llvm/include/llvm/Analysis/LoopInfo.h:388]: nit: `location -> debug location`?

File: llvm/lib/Transforms/Scalar/LoopDistribute.cpp [hunk 4]
```diff
@@ -686,6 +686,9 @@ class LoopDistributeForLoop {
     if (!Dependences || Dependences->empty())
       return fail("NoUnsafeDeps", "no unsafe dependences to isolate");
 
+    LLVM_DEBUG(dbgs() << "LDist: Found a loop: " << L->getHeader()->getName()
```
Comment (fhahn) [inline llvm/lib/Transforms/Scalar/LoopDistribute.cpp:689]: make clear that we found a loop LDIst can handle?

**Review by fhahn — COMMENTED** [review]
Can you add a test showing how the new debug info location is printed for new users?

**Comment by artagnon** [comment]
> Can you add a test showing how the new debug info location is printed for new users?

This is done in #93208.

**Comment by fhahn** [comment]
> > Can you add a test showing how the new debug info location is printed for new users?
> 
> This is done in #93208.

But that's separate from this patch, right? Still would be good to have a test here to check the behavior of the patch, could also be in LAA

**Comment by artagnon** [comment]
> > > Can you add a test showing how the new debug info location is printed for new users?
> > 
> > 
> > This is done in #93208.
> 
> But that's separate from this patch, right? Still would be good to have a test here to check the behavior of the patch, could also be in LAA

Not sure I follow. Isn't the debug location unstable (it's "<stdin>" or "<filename>" depending on whether the file was piped or if the file was passed in directly)?

I show the extra printed line in https://github.com/llvm/llvm-project/pull/93051/files#diff-691737e76443e1b38359d80e09fa9ad0cd14fd51ab4cc061d876604978fa4e2eR20-R21.

**Comment by fhahn** [comment]
> > > > Can you add a test showing how the new debug info location is printed for new users?
> > > 
> > > 
> > > This is done in #93208.
> > 
> > 
> > But that's separate from this patch, right? Still would be good to have a test here to check the behavior of the patch, could also be in LAA
> 
> Not sure I follow. Isn't the debug location unstable (it's "stdin" or "filename" depending on whether the file was piped or if the file was passed in directly)?
> 
> I show the extra printed line in https://github.com/llvm/llvm-project/pull/93051/files#diff-691737e76443e1b38359d80e09fa9ad0cd14fd51ab4cc061d876604978fa4e2eR20-R21.

I might be missing something, but wouldn't this need a test with debug location for the loop? Then the output should be stable?

File: llvm/lib/Analysis/LoopAccessAnalysis.cpp (hunk 2 above)
Comment (artagnon) [reply llvm/lib/Analysis/LoopAccessAnalysis.cpp:2406]: Unsure if we should change this output to diverge from LV.

File: llvm/lib/Transforms/Scalar/LoopDistribute.cpp (hunk 4 above)
Comment (artagnon) [reply llvm/lib/Transforms/Scalar/LoopDistribute.cpp:689]: Same here.

**Comment by artagnon** [comment]
> I might be missing something, but wouldn't this need a test with debug location for the loop? Then the output should be stable?

Fixed now, thanks for the review!

**Review by fhahn — APPROVED** [review]
LGTM, thanks!

**Pull request merged**